File: 백준/Bronze/13300. 방 배정/방 배정.js
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = require('fs')
  .readFileSync(path)
  .toString()
  .trim()
  .split('\n')
  .map((data) => data.split(' ').map(Number));

const [_, target] = input.shift();
let answer = 0;

const room = Array.from({ length: 6 }, () => [0, 0]);

input.forEach((student) => {
  const [gender, grade] = student;
  room[grade - 1][gender]++;
});

room.forEach((data) => {
  const numOfGirl = data[0];
  const numOfBoy = data[1];
  answer += Math.ceil(numOfGirl / target);
  answer += Math.ceil(numOfBoy / target);
});
console.log(answer);

File: 백준/Bronze/13300. 방 배정/방 배정.py
import sys, math

N, K = map(int, sys.stdin.readline().split())
mArr = [0] * 6
wArr = [0] * 6
answer = 0

for _ in range(N):
  S, Y = map(int, sys.stdin.readline().split())
  if (S == 0):
    wArr[Y - 1] += 1
  elif (S == 1):
    mArr[Y - 1] += 1
for w in wArr:
  if w > 0:
    answer += math.ceil(w / K)

for m in mArr:
  if m > 0:
    answer += math.ceil(m / K)

print(answer)